core/authors/views.py

```diff
@@ -184,4 +189,129 @@ def check_is_following(self, request, pk):
         
         return Response({
             "isFollowingUser": follow.exists()
-        }, status=200)
\ No newline at end of file
+        }, status=200)
+
+    # All posts the currently auth'd user can see of pk
+    # /author/{AUTHOR_ID}/posts
+    @action(detail=True, url_path="posts")
+    def posts(self, request, pk=None):
+        if request.user.is_anonymous:
+            # TODO should it be a text response?
+            return Response("You must be logged in to perform this action.",status=403)
+
+        if not pk:
+            # TODO should it be a text response?
+            return Response("You must specify an author.",status=400)
+        
+        requestingAuthor = request.user.author.id # Should be guaranteed because we checked above
+        page = int(request.query_params.get("page", 0)) + 1 # Must offset page by 1
+        if page < 1:
+            return Response("Page number must be positive", status=400)
+
+        # TODO: size should be limited?
+        size = int(request.query_params.get("size", 50))
+        if size < 0:
+            return Response("Size must be positive", status=400)
+
+        # post_types will track what level of posts a user can see
+        post_types = ["PUBLIC", "SERVERONLY"]
+        # convert to dict for dat O(1)
+        # Note: this is terrible, we should be using the database more directly
+        requesterFriends = {}
+        for friend in get_friends(requestingAuthor):
+            requesterFriends[friend] = True
+
+        # Check if they are direct friends
+        if requesterFriends.get(pk, False):
+            post_types += ["FRIENDS", "FOAF"]
+        else: # They are not direct friends, so we should check if they share any friends
+            for friend in get_friends(pk):
+                if requesterFriends.get(friend, False):
+                    post_types += ["FOAF"]
+                    break # we don't need to check any more friends
+
+        try:
+            posts = Posts.objects.all().filter(author=pk, visibility__in=post_types, unlisted=False)
```

Comment:
Images are meant to be unlisted, but they should still be returned from this endpoint I'm pretty sure. You may want to ask that on the forums

Reply:
Then you'll see two posts in this feed, the image and then the post the image is in?
Easy fix regardless

Reply:
@chaitali is more knowledgable about this than me but I think the idea is that we wouldn't display images that have unlisted set to true in the feed. We combine unlisted images along with the post they are part of

Reply:
Images will auto be combined with text posts on the frontend, I have that code in the current branch I'm working on.